packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/knowledge/KnowledgeController.java

```diff
@@ -260,29 +261,54 @@ public ResponseEntity<UUID> equationsToModel(
 			}
 		}
 
-		// Create a request for SKEMA with the cleaned-up equations.
-		final JsonNode skemaRequest = mapper.createObjectNode().put("model", "petrinet").set("equations", equationsReq);
-
-		// Get an AMR from Skema Unified Service
+		// Get an AMR from Skema Unified Service or MIRA
 		final Model responseAMR;
-		try {
-			responseAMR = skemaUnifiedProxy.consolidatedEquationsToAMR(skemaRequest).getBody();
-			if (responseAMR == null) {
-				log.warn("Skema Unified Service did not return a valid AMR based on the provided equations");
-				throw new ResponseStatusException(HttpStatus.UNPROCESSABLE_ENTITY, messages.get("skema.bad-equations"));
+		String extractionService = "mira";
+		if (req.get("extractionService") != null) {
+			extractionService = req.get("extractionService").asText();
+		}
+
+		if (extractionService.equals("mira")) {
+			final TaskRequest taskReq = new TaskRequest();
+			final String latex = req.get("equations").toString();
+			taskReq.setType(TaskType.MIRA);
+			try {
+				taskReq.setInput(latex.getBytes());
+				taskReq.setScript(LatexToAMRResponseHandler.NAME);
+				taskReq.setUserId(currentUserService.get().getId());
+				final TaskResponse taskResp = taskService.runTaskSync(taskReq);
+				final JsonNode taskResponseJSON = mapper.readValue(taskResp.getOutput(), JsonNode.class);
+				final String amrString = taskResponseJSON.get("response").asText();
+				ObjectNode objNode = (ObjectNode) mapper.readTree(amrString);
+
+				final JsonNode testNode = mapper.readValue(amrString, JsonNode.class);
+				responseAMR = mapper.convertValue(testNode, Model.class);
+			} catch (Exception e) {
+				e.printStackTrace();
```

Comment:
If we want a print out here it should use `log.error("some message", e);`

Reply:
done!